ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt

```diff
@@ -102,6 +109,19 @@ class ShopperInsightsClient @VisibleForTesting internal constructor(
         }
     }
 
+    private fun isTokenizationKey(): Boolean {
+        var isTokenizationKey = false
+        braintreeClient.getAuthorization { authorization, _ ->
```

Comment:
Not sure about Kotlin, but it looks like getAuthorization is in like completion handler and the boolean is being returned right away. Is this okay?

Reply:
Yes, it can be tested in the demo app to see the functionality by switching between client token and tokenization key

Reply:
Yep, I think we might have a bug here where the function returns before the lambda executes and sets `isTokenizationKey` to true.

One solution is to move the code below the tokenization key check in `getRecommendedPaymentMethods` to inside the lambda to ensure execution.

Reply:
Yeah I have a change locally for it but it's requiring some more major test changes because we use `lateinit` all over in our client tests currently. I may bug you a bit about it later if I can't get it sorted but have the check moved into the callback locally.

Reply:
Yeah won't need to do this in v5. It's only weird here because of the `AuthorizationLoader` pattern.

Reply:
We can implement a workaround though if we want to make a synchronous `BraintreeClient.hasTokenizationKeyAuth` property or something that's one possibility. Otherwise like Tim mentioned it could be done within the `getRecommendedPaymentMethods()` call.

Reply:
Updated here: https://github.com/braintree/braintree_android/pull/1057/commits/0e48fea6bfd932c504f0db759e6000d475dbae3f and https://github.com/braintree/braintree_android/pull/1057/commits/ba4553005e3d42e0ceb22b80057d3c7b1c4974eb. Thanks to @tdchow as always for being willing to answer my questions about how to do things in Kotlin with reguards to tests.